src/access/wac.ts

```diff
@@ -38,20 +45,97 @@ import {
 } from "../interfaces";
 import { internal_defaultFetchOptions } from "../resource/resource";
 import { Access } from "./universal";
-import { Access as AclAccess } from "../acl/acl";
+import {
+  Access as AclAccess,
+  AclDataset,
+  createAclFromFallbackAcl,
+  hasAccessibleAcl,
+  hasFallbackAcl,
+  hasResourceAcl,
+  saveAclFor,
+  WithAcl,
+  WithResourceAcl,
+} from "../acl/acl";
 
 // Setting WacAccess = Required<Access> enforces that any change on Access will
 // reflect on WacAccess, but WacAccess has additional restrictions.
-type WacAccess = (
+export type WacAccess = (
+  | { controlRead: true; controlWrite: true }
+  | { controlRead: false; controlWrite: false }
+  | { controlRead?: undefined; controlWrite?: undefined }
+) & {
+  read?: boolean;
+  append?: boolean;
+  write?: boolean;
+};
+
+type NoUndefinedWacAccess = (
   | { controlRead: true; controlWrite: true }
-  | { controlRead: undefined; controlWrite: undefined }
+  | { controlRead: false; controlWrite: false }
 ) & {
-  read: true | undefined;
-  append: true | undefined;
-  write: true | undefined;
+  read: boolean;
+  append: boolean;
+  write: boolean;
 };
 
-type AgentWacAccess = Record<WebId, WacAccess>;
+export type AgentWacAccess = Record<WebId, WacAccess>;
+
+function hasNoUndefinedAccessModes(
+  access: WacAccess
+): access is WacAccess & NoUndefinedWacAccess {
+  return !(
+    access.read === undefined ||
+    access.append === undefined ||
+    access.write === undefined ||
+    access.controlRead === undefined ||
+    access.controlWrite === undefined
+  );
+}
+
+function universalAccessToAcl(
+  newAccess: WacAccess,
+  previousAccess: AclAccess
+): AclAccess;
+function universalAccessToAcl(
+  newAccess: NoUndefinedWacAccess,
+  previousAccess: undefined
```

Comment:
Rou're still calling it with an explicit `undefined` value below, which seems kinda unnecessary.

```suggestion
```